tests/e2e/plugins/site-verification-api-mock.php

```diff
@@ -11,23 +11,48 @@
 
 register_activation_hook( __FILE__, function () {
 	delete_transient( 'gsk_e2e_site_verified' );
+	delete_transient( 'gsk_e2e_sc_site_exists' );
 } );
 register_deactivation_hook( __FILE__, function () {
 	delete_transient( 'gsk_e2e_site_verified' );
+	delete_transient( 'gsk_e2e_sc_site_exists' );
 } );
 
 add_action( 'rest_api_init', function () {
 
 	register_rest_route(
 		REST_Routes::REST_ROOT,
-		'modules/search-console/data/is-site-exist',
+		'modules/search-console/data/siteverification-list',
+		array(
+			'callback' => function () {
+				return array(
+					'type'       => 'SITE',
+					'identifier' => home_url( '/' ),
+					'verified'   => (bool) get_transient( 'gsk_e2e_site_verified' ),
+				);
+			}
+		),
+		true
+	);
+
+	register_rest_route(
+		REST_Routes::REST_ROOT,
+		'modules/search-console/data/siteverification',
```

Comment:
I just saw the overlap between this and new utilities in #400, like `setSiteVerification` and `setSearchConsoleProperty`. Whatever gets merged first, we can refactor the code accordingly after that to make use of them here.

Reply:
This is mocking a core API route, so not quite the same as our utilities 😄

Reply:
Hmm they both update the `googlesitekit_site_verified_meta` user meta though, so there is some overlap in the logic 🤔 
Maybe we can find a way. Otherwise, we leave it as-is 🙂